### DISCUSSION TITLE
Adapter Training for regression

### DISCUSSION THREAD

**Discussion by FahadEbrahim** [adapter-hub/adapters#637] [Q&A] [answered]
Hello, 

I'd like to create/train an adapter for a **Regression** task and the evaluation is based on MSE and Spearman's correlation coefficient. Any advice on what needs to be changed from a classification task in Adapter Training? 

With my thanks,
Fahad.

**Comment by calpt** [comment, marked as answer]
Hi @FahadEbrahim,

there should be not much that needs to be changed! The classification head provided by `AdapterModel` classes automatically switches to regression training (with MSE loss), if you set `num_labels` to 1, e.g.:

```python
model.add_classification_head(
    "task_name"
    num_labels=1,
)
```
The rest of the training script should be very similar to classification. As an example, you can have a look at the GLUE script, which supports STS-B (a regression task evaluated with spearmen coefficient) in addition to the other (classification) tasks: https://github.com/adapter-hub/adapters/blob/main/examples/pytorch/text-classification/run_glue.py

Hope this helps!
